core/input/input.cpp

```diff
@@ -72,6 +72,111 @@ static const char *_joy_axes[(size_t)JoyAxis::SDL_MAX] = {
 	"righttrigger",
 };
 
+static const char *_joy_button_generic_names[(size_t)JoyButton::SDL_MAX] = {
+	"Bottom Action",
+	"Right Action",
+	"Left Action",
+	"Top Action",
+	"Select",
+	"Guide",
+	"Start",
+	"Left Stick",
+	"Right Stick",
+	"Left Shoulder",
+	"Right Shoulder",
+	"D-pad Up",
+	"D-pad Down",
+	"D-pad Left",
+	"D-pad Right",
+	"Misc",
+	"Paddle 1",
+	"Paddle 2",
+	"Paddle 3",
+	"Paddle 4",
+	"Touchpad",
+};
+
+static const char *_joy_button_nintendo_names[(size_t)JoyButton::SDL_MAX] = {
+	"B",
+	"A",
+	"Y",
+	"X",
+	"-",
+	"",
+	"+",
+	"",
+	"",
+	"",
+	"",
+	"",
+	"",
+	"",
+	"",
+	"Capture",
+	"",
+	"",
+	"",
+	"",
+	"",
+};
+
+static const char *_joy_button_playstation_names[(size_t)JoyButton::SDL_MAX] = {
+	"Cross",
+	"Circle",
+	"Square",
+	"Triangle",
+	"Select",
+	"PS",
+	"",
+	"L3",
+	"R3",
+	"L1",
+	"R1",
+	"",
+	"",
+	"",
+	"",
+	"Microphone",
+	"",
+	"",
+	"",
+	"",
+	"",
+};
+
+static const char *_joy_button_xbox_names[(size_t)JoyButton::SDL_MAX] = {
+	"A",
+	"B",
+	"X",
+	"Y",
+	"Back",
+	"Home",
+	"Menu",
+	"L/LS",
+	"R/RS",
+	"LB",
+	"RB",
+	"",
+	"",
+	"",
+	"",
+	"Share",
+	"",
+	"",
+	"",
+	"",
+	"",
+};
+
+static const char *_joy_axis_names[(size_t)JoyAxis::SDL_MAX] = {
+	"Left Stick X-Axis",
+	"Left Stick Y-Axis",
+	"Right Stick X-Axis",
+	"Right Stick Y-Axis",
+	"Left Trigger",
+	"Right Trigger",
```

Comment:
These are `L2` and `R2` respectively on PlayStation, so maybe we should have a dedicated `_joy_axis_names` per vendor.